src/overcooked_ai_py/mdp/overcooked_env.py

```diff
@@ -68,17 +73,36 @@ def __init__(self, mdp_generator_fn, start_state_fn=None, horizon=MAX_HORIZON, i
                                             "If trying to instantiate directly from a OvercookedGridworld " \
                                             "instance, use the OvercookedEnv.from_mdp method"
         self.variable_mdp = _variable_mdp
-        self.mdp_generator_fn = mdp_generator_fn
+        # infinite case
+        if LST_LEN == -1:
+            self.mdp_generator_fn = mdp_generator_fn
+        # finite MDP LST case
+        else:
+            self.mdp_lst = [mdp_generator_fn() for _ in range(LST_LEN)]
+            self.mdp_generator_fn = lambda : random.choice(self.mdp_lst)
```

Comment:
I have a few questions about this approach:
1. how do we know how many mdps we are using before we start training?
2. the original design was designed with the option of curriculumn in mind (the mdp_generator_fn could carry curriculumn information in it, so the long term plan was to replace env.mdp_lst after the agents were doing well enough in the current layouts

Reply:
I'm not quite sure I understand your first question: don't you always know how many MDPs you are training on before starting training? To answer your question literally (don't think this is what you mean), you can know the number of layouts you're training on by just looking at the length of the mdp_list (which you know to be infinity if mdp_lst is none). 

Generally here I'm operating under the assumption that the Env is not changing during training: the mdp_generator_fn might be changing subtly over the course of training, but it will not be changing from like a single layout training to few-layout training, to infinite-layout training. In general, it's almost always a bad to train on few layouts at the beginning if eventually you're intending to train on infinitely many, because you will overfit.

For the changing of the generator fn during the course of training, we should figure out what APIs make sense to do that – you could either keep track of how many times the gen function was called within the Generator logic (in order to advance in the schedule), or maybe passing in some info to the generator function, rather than having it have no arguments).

In any case, I'm not sure why your curriculum should have any relationship to mdp_lst whatsoever?

Reply:
I see what you are saying. I think we can talk more about this later, but based on my observation of ppo learning on a finite set of layouts, the algorithm learns one layout at a time (basically, it will start with getting 0 on all the layouts. And then for 1 layout it will gradually learn to get up to ~100 sparse reward while all other layouts are still 0. And then it will start to conquer another layout, getting ~ 100 sparse reward, while the not-yet-learnt set of layout still all get 0). That's why I believe it might be good to dynamically change the size of mdp_lst in the future.

Reply:
But, I have made the change per your request, and it is on the newest commit